### ISSUE TITLE
ENV VARIABLES not reloading on lando rebuild or lando start in Laravel app

### ISSUE THREAD

**Issue by mkykode** [lando/lando#730]
**What is your lando version and operating system? **

v3.0.0-beta.35 on Mac OS High Sierra

```yml
name: blog
recipe: laravel
config:
  via: nginx
  database: mysql
  cache: redis
  xdebug: true
  webroot: public
```

**Tell us about the command you were running**

```bash
lando rebuild
lando restart
```

I setup a laravel application with the laravel recipe and used lando laravel new to set it up. Everything worked fine until I did lando rebuild or lando restart. When I execute these commands it assigns a new mysql port. I figured all I had to do is change the DB_PORT in the .env to the new port shown after running lando info, but that deosn't seem to do the trick. Eventhough the docs say that all lando will read the .env variables on there.

if I run php artisan serve, the application works as expected, however if I use the nginx urls assigned to the container it keeps the previous port from before the rebuild or restart and gives a SQL can't connect error. I tried all the laravel commands like lando php artisan config:clear or lando php artisan config:cache, but nothing works. I can't seem to get the domain to read the updated ENV variables on the .env file after the rebuild. It seems like it is caching the previous variables and I don't see any caching commands on the documentation. If I repeat the cycle again it will always update to the last rebuild set env variables.

What's weird is if I run lando config | grep DB_ it will show the correct port, but if I run lando php artisan tinker and then getenv('DB_PORT') it will show the previous port before the rebuild.

Am I missing something??? Why are the variables not being updated?

**Comment by dustinleblanc** [comment]
Hey @mkykode are you perhaps overriding this in you .env file? That will get loaded on top of whatever we're injecting into the container

**Comment by mkykode** [comment]
@dustinleblanc I expect that to be the behavior but it is not and exactly the issue I'm referring to. When i do lando rebuild, a new port gets assigned to the mysql container so I go in to the .env file to update it with the new port that lando info says it's assigned to. If I run a quick laravel server it register the new env file, but when running through lando (with the same domain url as before) it doesn't read the newly updated env file but the previous one before the rebuild.

**Comment by dustinleblanc** [comment]
I've got a laravel app on my laptop, I can try to repro with that, to clarify this is what I should do:

1. Start Lando and confirm life is good and Laravel app loads
2. `lando rebuild`
3. Check port assignments
4. Update .env file
5. Laravel won't load

One thing I've just realized, the *internal* port number should never change. For MySQL, it should always be 3306 inside the container, only the port you expose to localhost should get randomly assigned. For Postgres, it should be 5432. Does the app work if you leave the port as 3306?

**Comment by mkykode** [comment]
@dustinleblanc Maybe that's where I'm wrong, but I cannot seem to make the app work with the internal connection settings. So for example if I run:
```bash
mkdir test && cd test && lando init --recipe=pantheon && lando start && lando laravel new 
```
with the following .lando.yml
```yml
name: blog
recipe: laravel
config:
  via: nginx
  php: '7.2'
  database: mysql
  cache: redis
  xdebug: true
  webroot: public
```
Then I change the env file to have the variables as in your docs:
```env
APP_NAME=test
APP_ENV=local
APP_URL=http://test.lndo.site
DB_CONNECTION=mysql
DB_HOST=database
DB_PORT=3306
DB_DATABASE=laravel
DB_USERNAME=laravel
DB_PASSWORD=laravel
```
The app does start up, but the moment I try to do any database calls I get a SQL Connection error. The only way I get the db to calls to work is by changing the env vars to the external connection.

So , for example, if I create a Post controller & model with artisan:
```bash
lando php artisan make:model Post -m -r
```
and I try to send a variable to the view with a simple call like this inside the index method:
```php
public function index() 
{
$posts = Post::latest()->get();
return view('web', compact('posts'));
{
```
and on the view echo it out
```blade
@foreach ($posts as $post)
{{$post->title}}
@endforeach
```
I get Error:
Illuminate \ Database \ QueryException (2002)
SQLSTATE[HY000] [2002] Connection refused (SQL: select * from `posts` order by `created_at` desc)
<img width="642" alt="screen shot 2018-02-28 at 1 26 22 pm" src="https://user-images.githubusercontent.com/670135/36813950-059d7a74-1c8b-11e8-9481-cc5ec0a32a22.png">

Which made me think it was a connection issue so I changed it to the external randomly generated one, then the custom test.lndo.site url starts to work.
![screen shot 2018-02-28 at 1 08 18 pm](https://user-images.githubusercontent.com/670135/36813505-ad8001dc-1c89-11e8-8588-d82c4c962ce8.png)
to
"external_connection": {
      "host": "localhost",
      "port": "32793"
    },

To do this I have to lando restart or lando rebuild so it reads the new env file. It'll work the first time, but then if I do lando rebuild again it won't.

Am I missing a setting in lando?

**Comment by dustinleblanc** [comment]
no, nothing missing there, it should work using the first method, with all internal networking stuff. Out of curiousity, if you swap things over to postgres driver and use Postgres, do you get the same issue? I run my Laravel apps on Postgres and haven't run into this and I am guessing perhaps @pirog and @serundeputy might use PG too?

**Comment by tylerssn** [comment]
In the spirit of questioning the little things, is your Lando project root the same as your Laravel root? In other words, are they looking at the same `.env` files?

**Comment by mkykode** [comment]
@tylerssn yup it was. 
@dustinleblanc I threw away the project and started it over again making sure the database points to inside the container as we discussed and everything worked. I noticed if I change this, even if I revert and change it back to local it will give the errors from the beginning. It's almost like the local containers will override env variables with cache versions even after rebuild.

Thanks!
